packages/rs-dpp/src/data_contract/data_contract.rs

```diff
@@ -129,28 +130,24 @@ impl DataContract {
         let version = data_contract_map.get_u32(property_names::VERSION)?;
 
         // Defs
-        let defs = match data_contract_map.get("$defs") {
-            None => BTreeMap::new(),
-            Some(definition_value) => {
-                let definition_map = definition_value.as_map();
-                match definition_map {
-                    None => BTreeMap::new(),
-                    Some(cbor_map) => {
-                        let mut res = BTreeMap::<String, JsonValue>::new();
-                        for (key, value) in cbor_map {
-                            let key_string = key.as_text().ok_or_else(|| {
-                                ProtocolError::DecodingError(String::from(
-                                    "Expect $defs keys to be strings",
-                                ))
-                            })?;
-                            let json_value = cbor_value_to_json_value(value)?;
-                            res.insert(String::from(key_string), json_value);
-                        }
-                        res
-                    }
+        let defs = data_contract_map
+            .get("$defs")
+            .map(CborValue::as_map)
```

Comment:
I'm wondering why as_map doesn't fail, technically cbor value can be just a number (not sure). No need to do anything about this comment, just thinking out loud

Reply:
That's correct, it may fail, that's why we flatten next, because nested results have the same error type